Shared/Samples/Show device location with NMEA data sources/SimulatedNMEASentenceFeed.swift

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// A data source simulating a hardware that emits NMEA data.
+class SimulatedNMEASentenceFeed {
+    /// The playback speed multiplier.
+    private let playbackSpeed: Double
+    
+    /// An iterator to hold and loop through the mock NMEA data.
+    private var nmeaDataIterator: CircularIterator<Data>
+    
+    /// A timer to periodically provide NMEA data updates.
+    private var timer: Timer?
+    
+    /// An asynchronous stream of NMEA data.
+    var messages: AsyncStream<Data>!
+    
+    /// Loads locations from NMEA sentences.
+    /// Reads mock NMEA sentences line by line and group them by the timestamp.
+    /// - Parameters:
+    ///   - nmeaSourceFile: The URL of the NMEA source file.
+    ///   - speed: The playback speed multiplier.
```

Comment:
is this in seconds? if so the type should be `TimeInterval`

Reply:
Yeah I can make it second and call it interval instead of speed. Currently it is the reciprocal of time.